dask/array/tests/test_reductions.py

```diff
@@ -966,6 +966,17 @@ def test_quantile(rechunk, q, axis):
     )
 
 
+@pytest.mark.parametrize(
+    "func", [da.quantile, da.percentile, da.nanquantile, da.nanpercentile]
+)
+def test_quantile_func_family_with_axis_none(func):
+    # Check that these functions raise a
+    # NotImplementedError when axis=None
+    darr = da.ones((3, 3), chunks=(2, 2))
+    with pytest.raises(NotImplementedError):
+        func(darr, 0.5, axis=None)
+
+
```

Comment:
Could you also add a test here for the [else](https://github.com/dask/dask/pull/11988/files#diff-b60d0c0c32d24414e50ffbedb190c9e227b6b7db7e4727d1121b2e6bde68073fR1723) branch on L1724. I suspect that it isn't covered. Roughly

```suggestion
result = da.quantile(da.arange(5))
assert_eq(result, 2.0)


```

(untested)